hs_core/hydroshare/utils.py

```diff
@@ -163,10 +163,23 @@ def is_federated(homepath):
     Returns:
     True is the selected file indicated by homepath is from a federated zone, False if otherwise
     """
-    irods_storage = IrodsStorage('federated')
-    # if HS WWW iRODS proxy user can list homepath, homepath is federated; otherwise, it is not
-    # federated
-    return irods_storage.exists(homepath)
+    homepath = homepath.strip()
+    homepath_list = homepath.split('/')
+    if homepath_list[1]:
+        fed_proxy_path = os.path.join(homepath_list[1], 'home',
+                                      settings.HS_LOCAL_PROXY_USER_IN_FED_ZONE)
+        fed_proxy_path = '/' + fed_proxy_path
+    else:
+        # the test path input is invalid, return False meaning it is not federated
+        return False
+    if settings.REMOTE_USE_IRODS:
+        irods_storage = IrodsStorage('federated')
+    else:
+        irods_storage = IrodsStorage()
+
+    # if the iRODS proxy user in hydroshare zone can list homepath and the federation zone proxy
+    # user path, it is federated; otherwise, it is not federated
+    return irods_storage.exists(homepath) and irods_storage.exists(fed_proxy_path)
```

Comment:
@pkdash The setting ```REMOTE_USE_IRODS``` was added to accommodate some test VMs and local development environment in which irods federation is not set up, hence no federated iRODS Storage instance exists in these environments where REMOTE_USE_IRODS is set to False. Since this is_federated() function is used for these environments also, we will have to use a standard iRODS storage instance for these environments/cases, and irods_storage.exists(fed_proxy_path) will just return False in these cases, which is what we need. When the dockerized federated iRODS containers are used across the board for local development environments and for all test VMS, we will not need this REMOTE_USE_IRODS setting as well as a few other settings just for accommodating these environments where federation is not set up, and code can be simplified as well here and there once we get to that stage. I can take it on when that time comes. But for now, this code works well for both environments in which REMOTE_USE_IRODS is true or false.